packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -762,9 +762,16 @@ let dynamicImportedComponentCounter = 0;
 /**
  * create a dynamic component via `<x-foo lwc:dynamic={Ctor}>`
  */
-function dc(
+type LightningElementConfig = {
+    constructor: LightningElementConstructor;
+    props: Record<string, any>;
+};
+function isConfig(obj: any): obj is LightningElementConfig {
```

Comment:
A function called `isConfig` is too generic for this file. I would either rename it to `isLightningElementConfig` or inline it in the `dc`.